### ISSUE TITLE
[BUG]: Can’t update Anthropic or OpenAI api keys when workspace LLM preference differs from main app’s. 

### ISSUE THREAD

**Issue by atljoseph** [Mintplex-Labs/anything-llm#1440]
### How are you running AnythingLLM?

Docker (local)

### What happened?

Ollama is main configured provider.
I have a workspace with Anthropic configured.
When I initialized the connection it let me put in an api key. 
Can’t edit it any more…
The workaround is to go edit the main app settings to point at the specific provider where the api key needs to be updated. 
Then, update the api key and save. 
Then, change the provider back to Ollama, or something else and save. 
Finally, go back to the workspace and test. 
That’s a lot of steps. 
Seems like each provider should have sandboxed credential storage. 
if I was able to add the api key in a workspace, I should be able to edit it there. 

![IMG_1789](https://github.com/Mintplex-Labs/anything-llm/assets/25017621/3d4fc7c9-1fbe-4096-a541-ad2e66736b27)
![IMG_1788](https://github.com/Mintplex-Labs/anything-llm/assets/25017621/7b72aafc-ef0b-4542-ad02-658d754e54b4)

### Are there known steps to reproduce?

See description

**Comment by atljoseph** [comment]
Yeah I now realize this can be hard to work into the usership model. The best approach to this would be to define a provider settings screen that could apply to both workspace and globally. An admin user would initialize the env and open up this screen to input the top level api keys for each LLM provider they use. It could be both OpenAI and anthropic, for example. The admin could even gate which providers show up in workspace dropdowns by checking boxes next to each provider. Then, anytime a workspace uses one of those providers, the corresponding key is used. At the workspace level, a non-admin user would just override/augment the admin settings by plugging a different API Key into this screen, if they wish. In cases where the admin api key might have issues, it might be valuable (or on the other hand potentially destructive) for a non-admin to go in and update the workspace key. Use case: customer facing chat bot out of tokens and admin not in town

**Comment by nr-plaxon** [comment]
Having credentials per workspace would be a big improvement. We would love to use different Mistral accounts per workspace for example which is not possible right now.

**Comment by breesait** [comment]
Hello everyone, I've worked a PR to support this, I hope this is what you want(because I also want this feature )

**Comment by timothycarambat** [comment]
Moving to #4493 as this would fall under that as all providers would support this functionality and this would be solved by closing of that issue
